org.eclipse.lsp4e/src/org/eclipse/lsp4e/operations/linkedediting/LSPLinkedEditingReconcilingStrategy.java

```diff
@@ -94,13 +96,13 @@ public void install(ITextViewer viewer) {
 			super.install();
 			this.sourceViewer = thisViewer;
 			editorSelectionChangedListener = new EditorSelectionChangedListener();
-			editorSelectionChangedListener.install(sourceViewer.getSelectionProvider());
+			editorSelectionChangedListener.install(thisViewer.getSelectionProvider());
 		}
 	}
 
 	@Override
 	public void uninstall() {
-		if (sourceViewer != null) {
+		if (sourceViewer != null && editorSelectionChangedListener != null) {
```

Comment:
`EditorSelectionChangedListener` is still nullable, why do you remove the null check?

Reply:
I didnt remove the null check, I added it.

Reply:
I didnt remove the null check, I added it.